lib/pacemaker/pcmk_sched_native.c

```diff
@@ -2580,38 +2475,37 @@ LogActions(pe_resource_t * rsc, pe_working_set_t * data_set, gboolean terminal)
         if (next == NULL) {
         } else if ((migrate_op != NULL) && (current != NULL)
                    && pcmk_is_set(migrate_op->flags, pe_action_runnable)) {
-            LogAction("Migrate", rsc, current, next, start, NULL, terminal);
+            out->message(out, "transition", "Migrate", rsc, current, next, start, NULL);
 
         } else if (pcmk_is_set(rsc->flags, pe_rsc_reload)) {
-            LogAction("Reload", rsc, current, next, start, NULL, terminal);
+            out->message(out, "transition", "Reload", rsc, current, next, start, NULL);
 
 
         } else if (start == NULL || pcmk_is_set(start->flags, pe_action_optional)) {
             if ((demote != NULL) && (promote != NULL)
                 && !pcmk_is_set(demote->flags, pe_action_optional)
                 && !pcmk_is_set(promote->flags, pe_action_optional)) {
-                LogAction("Re-promote", rsc, current, next, promote, demote,
-                          terminal);
+                out->message(out, "transition", "Re-promote", rsc, current, next, promote, demote);
             } else {
                 pe_rsc_info(rsc, "Leave   %s\t(%s %s)", rsc->id,
                             role2text(rsc->role), next->details->uname);
```

Comment:
This output message is used in so many different ways that I think it's going to be tricky to merge this in.  At the least, I think I'll have to reproduce both the "eise if (start == NULL)..." condition as well as the "if (rsc->role == rsc->next_role)" condition surrounding it.  I'm fine doing that, if you don't think it's too much duplication.

Reply:
Not at all, that makes sense. The whole point of the message is to say what has to happen to get from the current role to the next role. Really, the entire LogActions() function starting with the first pe_rsc_info() call is the logical message.

Reply:
It might make sense to rearrange the code to reflect that, then.  The current transition_default and transition_xml functions (which need to be renamed, but whatever) could turn into helper functions that assemble the list_item out of all the various little pieces of data.  The bulk of the existing LogActions function then gets turned into the actual output message.

Reply:
And done.  I think I like the rearranged code better.  Next step is to rename all this transition stuff.